SpatialGDK/Source/SpatialGDK/Private/SpatialView/ScopedDispatcherCallback.cpp

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialView/ScopedDispatcherCallback.h"
+
+#include "SpatialView/Dispatcher.h"
+
+namespace SpatialGDK
+{
+FScopedDispatcherCallback::FScopedDispatcherCallback(FDispatcher& InDispatcher, const CallbackId InCallbackId)
+	: Dispatcher(&InDispatcher)
+	, ScopedCallbackId(InCallbackId)
+{
+	check(Dispatcher);
+}
+
+FScopedDispatcherCallback::~FScopedDispatcherCallback()
```

Comment:
With the way this is setup this type is unsafe to use outside of a unique_ptr, which is likely to make this used incorrectly later.

This should probably be explicitly deleting the copy constructor/assignment and defining a move constructor such that it either nulls out the dispatcher or sets the ID to 0. Then in the destructor we can check either of these conditions and only remove the callback if they are set. 

Alternatively we can add in a a unique_ptr internally although I'm not sure that works without making some heap allocation which seems excessive.

Reply:
Ah wait, I missed that this is explicitly deleing the move constructors too. Which I guess makes it safe, but also makes it a bigger pain that it needs to be. And means you don't use it as you'd intuitively expect.

Reply:
I was going to add a move constructor if/when we need it, I don't believe I need it at the moment.

Reply:
Happy to add one now if you prefer.

Reply:
I'd prefer to just do it now. I don't see any advantage to not doing it and it's only a few lines of code and it makes the use of the type much more intuitive.